distributed/utils.py

```diff
@@ -777,17 +778,30 @@ def _maybe_complex(task):
 
 
 def convert(task, dsk, extra_values):
-    if type(task) is list:
+    typ = type(task)
+    if typ is tuple and len(task) > 1 and type(task[0]) is SubgraphCallable:
+        sc = task[0]
+        return (
+            SubgraphCallable(
+                convert(sc.dsk, dsk, extra_values),
+                sc.outkey,
+                convert(sc.inkeys, dsk, extra_values),
+                sc.name,
+            ),
+        ) + tuple(convert(x, dsk, extra_values) for x in task[1:])
+    if typ is list:
         return [convert(v, dsk, extra_values) for v in task]
-    if type(task) is dict:
+    if typ is dict:
         return {k: convert(v, dsk, extra_values) for k, v in task.items()}
-    if istask(task):
+    if typ is tuple and task and callable(task[0]):  # istask(o)
         return (task[0],) + tuple(convert(x, dsk, extra_values) for x in task[1:])
```

Comment:
Maybe the first check and this one should be combined?  

```python
if type(task[0]) is SubgraphCallable:
    func = ...
else:
    func = task[0]

return (func,) + tuple(convert(...) for x in ...)
```

Reply:
Good point, fixed.